### PR TITLE
Implement edit list view

### PR DESCRIPTION
**Pull request by haiyingweng**
### Overview 
UI for edit list

### Changes Made
- Added `EditListViewController`
- Implemented select all and deselect 

🗒️ : I'll integrate backend for remove shows from list along with stuff from settings in the next PR. Also for move list functionality, it think the modal could be created in another PR since it could be reused for saving to list from the single media screen. 

### Related Issues
#15 

### Screenshots/GIFs
edit list

### CODE REVIEW

File: Flick/Assets.xcassets/deselect.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    },
+    {
+      "filename" : "deselectActive2x.png",
+      "idiom" : "universal",
+      "scale" : "2x"
+    },
+    {
+      "filename" : "deselectActive3x.png",
+      "idiom" : "universal",
+      "scale" : "3x"
```
Comment (haiyingweng) [inline Flick/Assets.xcassets/deselect.imageset/Contents.json:16]: i started putting 1x, 2x, 3x images cuz i think it's just better practice and most phones now use 3x so it's slightly costly to always upscale from 1x to 3x

Comment (lucyxubroad) [reply Flick/Assets.xcassets/deselect.imageset/Contents.json:16]: lol ya, I think this is good. youre required to do it anyways when you launch the app. I was just gonna do it all at once at the end lmao

Comment (lucyxubroad) [reply Flick/Assets.xcassets/deselect.imageset/Contents.json:16]: we also have to make sure all of our images are kind of consistent in sizing. I think they're all kinda all over the place right now.

Comment (lucyxubroad) [reply Flick/Assets.xcassets/deselect.imageset/Contents.json:16]: maybe ask Cindy to compile all of the icons with sizes somewhere for us.

File: Flick/Cells/MediaSeletableCollectionViewCell.swift [hunk 2]
```diff
@@ -66,6 +67,13 @@ class MediaSelectableCollectionViewCell: UICollectionViewCell {
         }
     }
 
+    func configure(media: Media) {
```
Comment (lucyxubroad) [inline Flick/Cells/MediaSeletableCollectionViewCell.swift:70]: we should eventually make this picture thing an function extension of UIImage haha

File: Flick/Controllers/EditListViewController.swift [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+        mediaCollectionView = UICollectionView(frame: .zero, collectionViewLayout: mediaCollectionViewLayout)
+        mediaCollectionView.backgroundColor = .white
+        mediaCollectionView.register(MediaSelectableCollectionViewCell.self, forCellWithReuseIdentifier: mediaCellReuseIdentifier)
+        mediaCollectionView.dataSource = self
+        mediaCollectionView.delegate = self
+        mediaCollectionView.showsVerticalScrollIndicator = false
+        mediaCollectionView.bounces = false
+        mediaCollectionView.allowsMultipleSelection = true
+        roundTopView.addSubview(mediaCollectionView)
```
Comment (lucyxubroad) [inline Flick/Controllers/EditListViewController.swift:117]: Do you think its possible to eventually make the media collection view its own component? it seems like its used a lot throughout the app but this is also only helpful if the components are largely the same. if there are too many little differences between then, its just a waste of time to make it its own component haha. just a thought. can make an issue out of this if you think its a good idea...

File: Flick/Controllers/EditListViewController.swift [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        setActionsActive(true)
+        numSelected += 1
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        numSelected -= 1
+        if numSelected == 0 {
+            setActionsActive(false)
```
Comment (lucyxubroad) [inline Flick/Controllers/EditListViewController.swift:259]: setActionsActive (numSelected!=0)

File: Flick/Controllers/EditListViewController.swift (hunk 3 above)
Comment (haiyingweng) [reply Flick/Controllers/EditListViewController.swift:117]: i agree with you. we def have quite many repetitive stuff in our code.

**Pull request merged**